src/windows/models/transition_model.py

```diff
@@ -58,6 +60,37 @@ def mimeData(self, indexes):
         return data
 
 
+class TransitionFilterProxyModel(QSortFilterProxyModel):
+    """Proxy class used for sorting and filtering model data"""
+
+    def filterAcceptsRow(self, sourceRow, sourceParent):
+        """Filter for common transitions and text filter"""
+
+        if get_app().window.actionTransitionsShowCommon.isChecked():
+            # Fetch the group name
+            index = self.sourceModel().index(sourceRow, 2, sourceParent) # group name column
+            group_name = self.sourceModel().data(index) # group name (i.e. common)
+
+            # Fetch the transitions name
+            index = self.sourceModel().index(sourceRow, 0, sourceParent) # transition name column
+            trans_name = self.sourceModel().data(index) # transition name (i.e. Fade In)
+
+            # Return, if regExp match in displayed format.
+            return "common" == group_name and self.filterRegExp().indexIn(trans_name) >= 0
+
+        # Continue running built-in parent filter logic
+        return super(TransitionFilterProxyModel, self).filterAcceptsRow(sourceRow, sourceParent)
+
+    def lessThan(self, left, right):
+        """Sort with both group name and transition name"""
+        leftData = self.sourceModel().data(left)
+        leftGroup = self.sourceModel().data(left, 257) # Strange way to access 'type' column in model
```

Comment:
@jonoomph 
I have to say, this terrified the living crap out of me, when I came across it while working on these. I absolutely _do not_ understand how it works, even **sort of**. "Strange" doesn't begin to cover it. Where did that even come from?

If I'm understanding it, though, the equivalent in non-magical Qt API should be something like:
```python
leftGroup = self.sourceModel().sibling(left.row(), 2, left).data()
# or even possibly
leftGroup = left.sibling(left.row(), 2).data()
```

Reply:
I'm almost tempted to think that it works because it's overflowing an old-school, single-byte `char` field.
(`255 + 2 = 257` after all.) Thus reaching the second column that way.

But that's still crazy, possibly even crazier than whatever's _actually_ happening here.

Reply:
@ferdnyc Ahhh, yes, that makes sense. I found this using a debugger (almost positive about that, because how else would I find the index... lol)

Reply:
@jonoomph Maybe we fix just this one, before the merge? :rofl: I'll look at fixing it now, in fact.

Reply:
Fixed. For a little extra flexibility, I made it respect `sortRole()` for the main column (only), instead of forcing the default `Qt.DisplayRole`. So the sort role can be changed, if needed at some future point, and it'll still work.

Reply:
Oops, looks like I was too slow by less than 5 minutes. :laughing: I'll do a new PR, nbd.

Reply:
Ooops, missed this comment.

Reply:
SON OF A **BITCH**!!!!

@jonoomph 

Five _months_ later, I finally discovered the source of those phantom `257` data items!

It seems that, unlike most of Qt's pre-packaged Model/View classes, `QStandardItem.data()` and `QStandardItem.setData()` both [**DEFAULT TO `Qt.UserRole + 1`**](https://doc.qt.io/qt-5/qstandarditem.html#setData) when not given an explicit role:

### `QVariant QStandardItem::data(int role = Qt::UserRole + 1) const`
> Returns the item's data for the given role, or an invalid `QVariant` if there is no data for the role.
> 
> Note: The default implementation treats `Qt::EditRole` and `Qt::DisplayRole` as referring to the same data.
> 
> See also `setData()`.

<hr>

### `void QStandardItem::setData(const QVariant &value, int role = Qt::UserRole + 1)`
> Sets the item's data for the given role to the specified value.
> 
> If you subclass `QStandardItem` and reimplement this function, your reimplementation should call `emitDataChanged()` if you do not call the base implementation of `setData()`. This will ensure that e.g. views using the model are notified of the changes.
> 
> Note: The default implementation treats `Qt::EditRole` and `Qt::DisplayRole` as referring to the same data.
> 
> See also `Qt::ItemDataRole`, `data()`, and `setFlags()`.

I guess those handwaves about "The default implementation treats..." are their explanation for it, but it's still a pain in the ass. Classes like `QAbstractItemModel`— and, hell, evne `QStandardItemModel`! — return the _actual_ `Qt.DisplayRole` and `Qt.EditRole` for `data()` and `setData()` respectively.

So, the long and short of it is, **WE'RE** the ones shoving all of those role-257 data items into the data models, anytime we call `setData()` or `data()` on an instance of `QStandardItem`!

(And now that I know this, I can adjust my user-defined roles to start from _higher_ than `Qt.UserRole + 1`. Which would've been really nice to know like 5 or 6 hours ago, insead of spending them pointlessly fighting with disappearing model data. 😡)

Reply:
This is entirely @N3WWN 's fault. Stupid properties table keyframe-status code!